macros/materializations/dummy.sql

```diff
@@ -0,0 +1,10 @@
+{# Materialization that does not create any table at the end of its run #}
+{# An example for a usage case is when we want a model to appear on the Elementary lineage graph, but the table is created outside of dbt #}
+{% materialization dummy, default -%}
```

Comment:
I'm not 100% sure about the name from a product perspective - since we're going to generate a bunch of these for users maybe it should say something like "external" or "non_dbt"?
Worth consulting with @oravi , the king of names :)

Though I agree that functionality-wise "dummy" makes sense.